### PR TITLE
Added support in Version upgrade test framework to test upgrade with non-default-server-collation

### PR DESCRIPTION
**Pull request by rohit01010**
### Description
This pull request adds a support in version upgrade test framework to perform a upgrade with non-default server collation name. This pull request also add normal jdbc with non-default server collation workflow which will run all the jdbc tests with non-default server collation name.

### Upgrade paths added to test upgrade with non-default server collation
```
[14.6 --> 15.latest] with server collation name: chinese_prc_ci_as
[14.6 --> 15.latest] with server collation name: japanese_ci_as
[14.11 --> 15.latest] with server collation name: chinese_prc_ci_as
[13.6 --> 14.11 --> 15.latest] with server collation name: chinese_prc_ci_as
[14.3 --> 14.11 --> 15.latest] with server collation name: chinese_prc_ci_as
```

### JDBC Test with non-default server collation
Currently in this JDBC Test with non-default server collation, JDBC tests will run with server collation as  `chinese_prc_ci_as`. (Note: server collation can be changed by modifying `SERVER_COLLATION_NAME` env variable to desired server collation in workflow file for this test)

Signed-off-by: Rohit Bhagat [rohitbgt@amazon.com](mailto:rohitbgt@amazon.com)

### Issues Resolved

BABEL-4590

### Test Scenarios Covered ###
* **Use case based -** NA

* **Boundary conditions -** NA

* **Arbitrary inputs -** NA

* **Negative test cases -** NA

* **Minor version upgrade tests -** NA

* **Major version upgrade tests -** NA

* **Performance tests -** NA

* **Tooling impact -** NA

* **Client tests -** NA

### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is under the terms of the Apache 2.0 and PostgreSQL licenses, and grant any person obtaining a copy of the contribution permission to relicense all or a portion of my contribution to the PostgreSQL License solely to contribute all or a portion of my contribution to the PostgreSQL open source project.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: .github/composite-actions/run-pg-upgrade/action.yml [hunk 1]
```diff
@@ -32,12 +36,17 @@ runs:
...
+        sudo ~/${{ inputs.pg_new_dir }}/bin/psql -d jdbc_testdb -U runner -c "ALTER EXTENSION babelfishpg_common UPDATE;"
+        sudo ~/${{ inputs.pg_new_dir }}/bin/psql -d jdbc_testdb -U runner -c "ALTER EXTENSION babelfishpg_tsql UPDATE;"
         sudo ~/${{ inputs.pg_new_dir }}/bin/psql -d jdbc_testdb -U runner -c "\dx"
         echo 'Reset bbf database settings...'
         sudo ~/${{ inputs.pg_new_dir }}/bin/psql -d jdbc_testdb -U runner -c "ALTER SYSTEM SET babelfishpg_tsql.database_name = 'jdbc_testdb';"
         sudo ~/${{ inputs.pg_new_dir }}/bin/psql -d jdbc_testdb -U runner -c "ALTER SYSTEM SET babelfishpg_tsql.migration_mode = '${{inputs.migration_mode}}';"
         sudo ~/${{ inputs.pg_new_dir }}/bin/psql -d jdbc_testdb -U runner -c "SELECT pg_reload_conf();"
+        if [[ ${{ inputs.server_collation_name }} != "default" ]]; then
+          sudo echo "babelfishpg_tsql.server_collation_name = '${{ inputs.server_collation_name }}'" >> ~/${{ inputs.pg_new_dir }}/data/postgresql.conf
+          ~/${{ inputs.pg_new_dir }}/bin/pg_ctl -D ~/${{ inputs.pg_new_dir }}/data/ -l ~/${{ inputs.pg_new_dir }}/data/logfile14 restart
```
Comment (thephantomthief) [inline .github/composite-actions/run-pg-upgrade/action.yml]: why do we need to hardcode logfile14?

Comment (rohit01010) [reply .github/composite-actions/run-pg-upgrade/action.yml]: Same log file was used above to start the server, hence I used same log file restart.

Comment (rohit01010) [reply .github/composite-actions/run-pg-upgrade/action.yml]: Updated the hardcoded logfile name to generic name

**Review by thephantomthief — APPROVED** [review]

File: test/JDBC/src/test/java/com/sqlsamples/TestQueryFile.java [hunk 2]
```diff
@@ -430,6 +431,12 @@ public void TestQueryBatch(String inputFileName) throws SQLException, ClassNotFo
         File expectedFile;
         if (isParallelQueryMode && checkParallelQueryExpected)
             expectedFile = new File(parallelQueryGeneratedFilesDirectoryPath + outputFileName + ".out");
+        else if(serverCollationName != "default"){
```
Comment (basasairohan) [inline test/JDBC/src/test/java/com/sqlsamples/TestQueryFile.java]: what happens when we run in parallel query mode with non-default server collation name?

Comment (rohit01010) [reply test/JDBC/src/test/java/com/sqlsamples/TestQueryFile.java]: Execution remains same, if some query has different expected output in parallel query mode with non-default server collation compared to output in parallel query mode with default server collation then that test will fail with the corresponding output diff. Currently no support is added to provide different output in the above scenario.

Comment (rohit01010) [reply test/JDBC/src/test/java/com/sqlsamples/TestQueryFile.java]: Added support to provide different output in the above scenario.

**Review by basasairohan — APPROVED** [review]

File: test/JDBC/README.md [hunk 3]
```diff
@@ -415,3 +416,29 @@ After building the modified PostgreSQL engine and Babelfish extensions using the
...
+   ```bash
+    export serverCollationName=<server_collation_name>
+    # Verify if serverCollationName is set to correct collation name
+    echo $serverCollationName
+   ```
+3. Now Run the tests:
+    ```bash
+    mvn test
+    ```
+4. If the expected output is different when run with non-default server collation, one can add a different expected output in `expected/non_default_server_collation/<server_collation_name>` folder. Similary, if tests are running in parallel query mode and expected output is different with non-default server collation, one can add a different expected output in `expected/parallel_query/non_default_server_collation/<server_collation_name>/` folder.
```
Comment (basasairohan) [inline test/JDBC/README.md]: This seems a bit confusing. If the tests are run in parallel query mode, there might be two cases right? 

1. Difference in output due to parallel query mode and non default server collation, in this case we add the expected output in `expected/parallel_query/non_default_server_collation/<server_collation_name>/`
2. Difference in output only due to non default server collation, we add it in `expected/non_default_server_collation/<server_collation_name>`

The above point does not represent this. Can you please update it ?

**Review by basasairohan — CHANGES_REQUESTED** [review]
Please introduce a normal JDBC workflow with non-default-server-collation

File: test/JDBC/README.md (hunk 3 above)
Comment (rohit01010) [reply test/JDBC/README.md]: If the tests are run in parallel query mode, two cases are as follows
case 1: parallel query enabled with default server collation
- adding different expected output in this case remains same [Reference](https://github.com/babelfish-for-postgresql/babelfish_extensions/tree/BABEL_3_X_DEV/test/JDBC#running-tests-with-parallel-query-enabled).

case 2: parallel query enabled with non-default server collation
- adding different expected output in this case are added in the readme

Anyways I'll add all cases explicitly

Comment (rohit01010) [reply test/JDBC/README.md]: Done

**Review by basasairohan — APPROVED** [review]

**Review by Deepesh125 — APPROVED** [review]

**Pull request merged**